torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1250,10 +1250,11 @@ void LoopNest::splitWithMask(For* f, int factor, For** outer, For** inner) {
   body_inner = Substitute(body_inner, {{f->var(), combined_index}});
 
   *inner = new For(i_inner, new IntImm(0), factor_expr, body_inner);
-  *outer =
-      new For(i_outer, new IntImm(0), split_count, *inner, f->loop_options());
-
-  p->replace_stmt(f, *outer);
+  f->setVar(i_outer);
```

Comment:
We' re still cloning `body_inner` on line 1236. Can we avoid that? Otherwise, these loops would be invalidated:

Before Split:
```
for i in 0..N:
  for j in 0..100:
   ...
```

After Split:
```
for io in 0..8:
  for ii in 0..N/8:
    for j_new in 0..100: # << this is not the same loop as loop 'j'
     ...
```

Can we also add a test for such case?

Reply:
Good catch with the cloning of `body_inner`. I removed that in `splitWithTail` but missed it here. Fixed.

However, that is not enough to ensure that the body remains the same. We need to fix the `IRMutator` APIs as well as `Substitute` for that. I'm starting to work on that. Once that is fixed, I will add a test here to check the same.